### DISCUSSION TITLE
v34.0-beta1 Pre-Release Feedback

### DISCUSSION THREAD

**Discussion by peternewman** [streetcomplete/StreetComplete#3216] [Feedback]
Some assorted feedback (apologies if you'd rather have it in a discussion or elsewhere or separate tickets, I couldn't see something linked already):
Quest presets
- It would be nice to be able to rename presets (perhaps press and hold)
- Alternatively/as well, it would be nice to be able to tap the preset name to suggest it
- I thought one of the previews showed the preset name in the main hamburger menu. I'd appreciate this to see where I am, even if it only shows up when you're off default to avoid confusing 
- Just noticed, the selected radio button animation seems quite slow to me
- Perhaps show/hide preset name in the Settings menu based on the number of presets (>1) rather than whether we're on the default or not
- Although conversely would it be nice to ship with a few additional demo presets (e.g. in a vehicle, indoor first)

Is Lit Quests
- It was dark for me and some quests weren't showing, nothing obvious showed in the preset list. I think I went into the quests menu and probably restarted the app and some started appearing. I don't know if this was a bug or just the changing timing. The suggestion/mention to show when it would become available would be great.
- Assuming it's not a bug, the time seems later than necessary to me. In the UK lighting up time is legally defined ( https://en.wikipedia.org/wiki/Lighting-up_time ), but I accept you may want to add a buffer to guarantee, or it defeats the purpose of hiding the quests.
- Personally I'd still like to be able to override it, which I thought was previewed at one point.

General
- I seem to be getting lots of extra crossing quests around marked cycle lanes e.g. around here https://www.openstreetmap.org/#map=19/51.52374/-0.13026 some were valid, but some weren't (e.g. it was assuming three crossings for a road with two cycleways).
- The overwhelming quests message appears after the inbox notification, I wonder if it should be the other way round given it's just a one off.
- I've got a few more ease of answering suggestions, but I need to check I didn't make them before and they got ruled out, so look out for a PR.

Overall it looks good and I'm looking forward to sorting out my presets properly and using them thanks! :heart:

**Comment by smichel17** [comment]
I took the liberty to convert this to a discussion, I think a grab bag of stuff is a better fit here (in "feedback" :slightly_smiling_face:)  
edit: in retrospect, this was probably a bit impulsive, but what's done is done.

**Reply by peternewman** [reply]
No, it's a fair point. it does mean we can have a thread for each "issue" without flooding the issue tracker.

**Comment by matkoniecz** [comment]
> Personally I'd still like to be able to override it, which I thought was previewed at one point.

It was proposed at some point by me - but it was after code was already written (I thought that PR is stale and I will write new one, possibly reusing existing one)

**Reply by westnordost** [reply]
It wasn't previewed and it is not planned to be implemented.

**Reply by peternewman** [reply]
I'd certainly appreciate it given the behaviour I saw and how I used to use it. I think it was just @matkoniecz I'd seen then.

**Comment by matkoniecz** [comment]
> I seem to be getting lots of extra crossing quests around marked cycle lanes e.g. around here https://www.openstreetmap.org/#map=19/51.52374/-0.13026 some were valid, but some weren't (e.g. it was assuming three crossings for a road with two cycleways).

Can you link specific one? For example https://www.openstreetmap.org/node/108006 (I have not tested is this one or other generating invalid crossing quest)

**Reply by westnordost** [reply]
I downloaded that area, and not a single quest of this type was displayed. @peternewman , I need feedback from you here.

**Reply by peternewman** [reply]
See for example the reality:
https://www.google.com/maps/@51.5238747,-0.1297619,3a,75y,245.18h,67.66t/data=!3m6!1e1!3m4!1segyLKKcQ5sR6wZECLruuiA!2e0!7i13312!8i6656

Maybe it's more of a mapping thing as I see they are all mapped as crossings:
https://www.openstreetmap.org/node/5610095005
https://www.openstreetmap.org/node/485257348
https://www.openstreetmap.org/node/485257346

Or perhaps map style wise it would help if you could differentiate roads from cycleways from footpaths as sometimes I'm walking on the pavement/sidewalk of a road, sometimes I'm on a separately mapped pavement path and in cases like this I'm walking on an unmapped pavement path, then there's a mapped cycleway (which in this case is attached to the road on one side and not on the other) and then there's a mapped road.

I assumed this was due to the new places to cross quest, but its probably not from the looks of things, although it is still very confusing to answer. See also the does this crossing have an island!?!

**Comment by westnordost** [comment]
> Assuming it's not a bug, the time seems later than necessary to me. In the UK lighting up time is legally defined ( https://en.wikipedia.org/wiki/Lighting-up_time ), but I accept you may want to add a buffer to guarantee, or it defeats the purpose of hiding the quests.

StreetComplete uses [Civil Twilight ](https://en.wikipedia.org/wiki/Twilight#Civil_twilight) which exactly matches the rules in the article you mentioned.

**Reply by peternewman** [reply]
Hmm, then maybe the street lights are switched on well before that (about an hour). I'll try and have a better look today.

**Reply by peternewman** [reply]
Okay, so my nearby street lights turned on about 20 minutes before Civil Twilight (as it had got dark).

However I think I've found the bigger issue, even an hour later, and despite panning and zooming around and switching to other apps, the lit quests still hadn't appeared. They did the moment I used Scan for Quests Here in the hamburger menu though. That's a bit counter-intuitive to me, I assumed they'd just pop up at the right time.

Also, and I haven't tested properly, if I'm using SC offline and I try and scan, it says I can't as I'm offline. I'm assuming this means that workaround won't work to recover them for me. Although maybe changing presets or relaunching the app will do the trick? If I realise it's got dark enough and remember to do so.

**Reply by westnordost** [reply]
What if you solve a single quest? Do they not pop in then?

**Comment by westnordost** [comment]
> The overwhelming quests message appears after the inbox notification, I wonder if it should be the other way round given it's just a one off.

Ok

**Comment by westnordost** [comment]
> Perhaps show/hide preset name in the Settings menu based on the number of presets (>1) rather than whether we're on the default or not

Good idea, will do

**Comment by westnordost** [comment]
> It would be nice to be able to rename presets (perhaps press and hold)
> I thought one of the previews showed the preset name in the main hamburger menu. I'd appreciate this to see where I am, even if it only shows up when you're off default to avoid confusing
> Although conversely would it be nice to ship with a few additional demo presets (e.g. in a vehicle, indoor first)

Not planned. The renaming I would accept as a PR, however it IMO should be just a ✏️ icon left of the delete button.

**Reply by peternewman** [reply]
Just the rename? You don't think it should show the preset in the main hamburger (again only if >1 preset)?

Fair point on the pencil, that would be just as obvious.

You don't think some demo presets are worthwhile?

**Reply by smichel17** [reply]
imo it should show (and allow changing) the preset in the main hamburger… but, also imo, this is a fine thing to add (perhaps by someone else) after the initial feature is released.

**Comment by peternewman** [comment]
Also the debugging aspect of presets has worked great for me, I can create a test preset, turn everything off and just enable the quest(s) I want to look at, so much less faff than what I had to do before.

**Reply by smichel17** [reply]
Ref: https://github.com/streetcomplete/StreetComplete/issues/3228

**Comment by peternewman** [comment]
> * it would be nice to be able to tap the preset name to suggest it

Pretty please. Even now I keep just tapping it and wondering why it didn't change. It would also match the behaviour of the switches for "Keep screen on" and "Select theme" etc. Although I see Quest selection doesn't do it either, but that's never been an issue for me (maybe the fact most are ticked makes the checkboxes more eye-catching).

**Comment by peternewman** [comment]
> * I've got a few more ease of answering suggestions, but I need to check I didn't make them before and they got ruled out, so look out for a PR.

See #3218
